pkg/snapshot/README.md

```diff
@@ -0,0 +1,203 @@
+3. Credentials to connect Tyk Dashboard. Please visit [Tyk Docs](https://tyk.io/docs/tyk-stack/tyk-operator/installing-tyk-operator/#tyk-self-managed-hybrid) for details.
+
+## Installation
+
+In order to use snapshot tool, checkout "feat/importer-cli" branch from Tyk Operator repository on your local machine.
+
+1. Clone Tyk Operator repository
+```bash
+git clone https://github.com/TykTechnologies/tyk-operator.git
+cd tyk-operator/
+git checkout feat/importer-cli
+```
+
+2. Build Tyk Operator
+```bash
+go build
+```
+
+## Preparation
+1. Use API Category to group the APIs you want to export
+
+By default, snapshot tool will export all APIs from Tyk dashboard with category `operator`.  You can also categorize your APIs by teams, environments, or any way you want. You can specify which API category to be exported in later steps.
+
+![apis](./img/apis.png)
+
+2. Specify the Kubernetes resource names in Config Data
+
+By default, the created ApiDefinition resource will have the name `replace-me-{i}`. It is not the best name for a kubernetes resource and should be manually updated. You can store this meta-data in Config Data for each API first before exporting, so that you do not need to manually update the output file afterwards.
+
+![config-data](./img/config-data.png)
+
+## Usage
+```bash
+tyk-operator exports APIs and Security Policies from your Tyk Dashboard to Custom Resource that can be used with Tyk Operator
+
+Export API Definitions:
+   --snapshot <output_file>
+    	Pull a snapshot of ApiDefinitions from Tyk Dashboard and output as CR
+  --all
+    	Dump all APIs
+  --category <category_name>
+    	Dump APIs from specified category (default "operator")
+
+Export Security Policies:
+  --policy <output_file>
+    	Pull a snapshot of SecurityPolicies from Tyk Dashboard and output as CR
+```
+
+
+
+### Setting required environment variables
+
+Store the Tyk Dashboard connection parameters in environment variables before running `tyk-operator`, e.g.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> \
+./tyk-operator --snapshot <OUTPUT_FILE>
+```
+
+where
+- `<TYK_URL>`: Management URL of your Tyk Dashboard.
+- `<TYK_AUTH>`: Operator user API Key.
+- `<TYK_ORG>`: Operator user ORG ID.
+
+> For more details on how to obtain the URL and credentials, please visit [Tyk Docs](https://tyk.io/docs/tyk-stack/tyk-operator/installing-tyk-operator/#tyk-self-managed-hybrid).
+
+
+
+### Exporting API Definitions
+
+#### Specify Category to export
+
+By default, tyk-operator exports ApiDefinitions defined in `#operator` category. You can change default category via `--category` flag.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --category k8s
+```
+
+You can also dump all ApiDefinitions from Dashboard via `--all` flag.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --all
```

Comment:
Rather than `<TYK_URL>`, I would prefer `${TYK_URL}` that way the snippet can be copied and pasted - and it would just work if we have an env var.